services/clusters-keeper/src/simcore_service_clusters_keeper/modules/clusters_management_core.py

```diff
@@ -49,7 +54,16 @@ async def _find_terminateable_instances(
             # let's terminate that one
             terminateable_instances.append(instance)
 
-    return terminateable_instances
+    # get all terminateable instances associated worker instances
+    worker_instances = []
+    for instance in terminateable_instances:
+        user_id = instance.tags["user_id"]
+        wallet_id = instance.tags["wallet_id"]
```

Comment:
Q: what if the tags are missing, this will raise an error, should it not be handled gracefully?

Reply:
currently impossible. but I see the point. will make it explicit.

Reply:
currently impossible. but I see the point. will make it explicit.